test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -448,7 +448,7 @@ export default class PerfCompare extends Component {
                     metricProps = this.metricsProps[benchmark][curMetricName];
                 }
                 // get metric Properties (regex & higherbetter & units) using current benchmark information
-                const curMetricUnits = metricProps ? this.metricsProps.units : "";
+                const curMetricUnits = metricProps ? metricProps.units : "";
```

Comment:
Should we also check for `metricProps.units`?

Reply:
In definition for benchmarkMetrics, it is required to add below three elements(units is required).
```
Format of each Benchmark
* 	Benchmarkname: {
* 	outerRegex:  Regex with capture group: all text(*) after warmup runs should be used
*  NOTE : Outer regex is optional param should be only used when there are cold runs 
*  that you do not wish to include as result
*  metrics:
*    {  //  metrics object with following properties.
*       name of metric : { //individual metric 
*           regex: Regex with capture group: value for the specific metric
*           higherbetter: variable to dettermine if its better to have higher value
*           units: unit of measurement
*           (Optional) funcName: storing a function that will run on the current set of metric values.
*                                Metrics such as JITCPU total in LibertyThroughput and geomean_GCA in CryptoBB require aggregate function to be applied. 
*                                e.g JIT CPU total values are collected as [ 0,0,300,200,20 ] but we display 520 as the JITCPU total  
*
*        }, and more metrics in same format...
```

Reply:
There is no check to enforce this on the server side. Until it is implemented on the server side, I would prefer to have the check here

Reply:
added as required